src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -289,28 +290,46 @@ public void parseRequestPathAndQueryTest() throws IOException, URISyntaxExceptio
 	public void listenerReconnectionTest() {
 		AtomicInteger handlerExecuted = new AtomicInteger(0);
 		AtomicReference<Throwable> exception = new AtomicReference<>(new Exception());
-		
-		listener.setConnectingHandler((ex) -> {
+		listener.setConnectingHandler(ex -> {
 			exception.set(ex);
 			handlerExecuted.incrementAndGet();
 		});
 		
 		listener.openAsync(Duration.ofSeconds(15)).join();
-		ControlConnection controlConnection = listener.getControlConnection();
-		controlConnection.getConnectAsyncTask().thenCompose(controlWebSocket -> {
-			return controlWebSocket.closeAsync(); // This bypasses listener.close(), simulating an unexpected close
-		}).join();
-		
+		listener.injectFault(new ConnectionLostException());
+		assertFalse("listener should be disconnected temporarily for now", listener.isOnline());
 		CompletableFuture<Void> reconnectTask = CompletableFutureUtil.delayAsync(Duration.ofMillis(1000), EXECUTOR).thenRun(() -> {
 			assertTrue("listener should be reconnected now", listener.isOnline());
 		});
 
-		assertFalse("listener should be disconnected temporarily for now", listener.isOnline());
-		assertEquals("The reconnecting handler was not called exactly once", 1, handlerExecuted.get());
+		assertEquals("The reconnecting handler was not called exactly once.", 1, handlerExecuted.get());
 		assertTrue("Exception should be ConnectionLostException.", exception.get() instanceof ConnectionLostException);	
 		reconnectTask.join();
 	}
 	
+	// The listener should continuously try to reconnect if network is lost
+	@Test
+	public void listenerNetworkLostTest() throws Exception {
+	    int delayIndex = 2;
```

Comment:
>2 [](start = 22, length = 1)

Should work for all values 2-5, I just wanted the test to take the minimal time

Reply:
Not needed for this check-in but I'm curious if there are ways (events or similar) to make it catch the online handler as soon as it happens instead of sleeping long enough that it should happen no matter what...

---
In reply to: [354519489](https://github.com/Azure/azure-relay-java/pull/58#discussion_r354519489) [](ancestors = 354519489)